pengine/allocate.c

```diff
@@ -1421,11 +1422,15 @@ stage6(pe_working_set_t * data_set)
                 dc_down = stonith_op;
                 dc_fence = stonith_op;
 
-            } else {
+            } else if (is_set(data_set->flags, pe_flag_concurrent_fencing) == FALSE) {
                 if (last_stonith) {
                     order_actions(last_stonith, stonith_op, pe_order_optional);
                 }
                 last_stonith = stonith_op;
+
+            } else {
+                order_actions(stonith_op, done, pe_order_implies_then);
+                stonith_ops = g_list_append(stonith_ops, stonith_op);
```

Comment:
What's the benefit of requiring an explicit concurrent-fencing option rather than always using it, or using it automatically when any device has pcmk_action_limit > 1? It seems complicated and error-prone on the user's end.